### PR TITLE
fix(sql): support pg_attribute and pg_database pg pseudo-tables with no schema prefix

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Ref: #3033

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgDatabaseFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgDatabaseFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgAttributeFunctionFactory.java|2|2|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgAttributeFunctionFactory.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
 public class PgAttributeFunctionFactory implements FunctionFactory {
 
     private static final RecordMetadata METADATA;
-    private static final String SIGNATURE = "pg_catalog.pg_attribute()";
+    private static final String SIGNATURE = "pg_attribute()";
```
Comment (eugenels) [inline core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgAttributeFunctionFactory.java:46]: Do we need to apply the same technique to other functions? We still have several functions that only have a prefix.

Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgAttributeFunctionFactory.java:46]: Good finding. Tried to make them more consistent in d4791f7e3eff6a2f4eb01fc29cf57feedaabe25d

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 274 / 285 (96.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/CurrentSchemasFunctionFactory.java|7|13|53.85%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgIndexFunctionFactory.java|11|12|91.67%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgRolesFunctionFactory.java|20|21|95.24%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgLocksFunctionFactory.java|23|24|95.83%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgGetKeywordsFunctionFactory.java|30|31|96.77%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgClassFunctionFactory.java|148|149|99.33%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgIsInRecoveryFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/TxIDCurrentFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgGetPartKeyDefFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgClassFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedCurrentSchemasFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgRolesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgDatabaseFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgGetKeywordsFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedTxIDCurrentFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgDatabaseFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgGetSITExprFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgGetSITExprFunctionFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgLocksFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgGetPartKeyDefFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgIsInRecoveryFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgIndexFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgNamespaceFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgAttributeFunctionFactory.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 274 / 285 (96.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/CurrentSchemasFunctionFactory.java|7|13|53.85%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgIndexFunctionFactory.java|11|12|91.67%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgRolesFunctionFactory.java|20|21|95.24%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgLocksFunctionFactory.java|23|24|95.83%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgGetKeywordsFunctionFactory.java|30|31|96.77%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgClassFunctionFactory.java|148|149|99.33%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgIsInRecoveryFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/TxIDCurrentFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgGetPartKeyDefFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgClassFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedCurrentSchemasFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgRolesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgDatabaseFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgGetKeywordsFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedTxIDCurrentFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgDatabaseFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgGetSITExprFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgGetSITExprFunctionFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgLocksFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgGetPartKeyDefFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgIsInRecoveryFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgIndexFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgNamespaceFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgAttributeFunctionFactory.java|2|2|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgGetKeywordsFunctionFactory.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            @Override
+            public int getStrLen(int col) {
+                if (col == 0) {
+                    return getStr(col).length();
+                }
+                return -1;
+            }
+        }
+    }
+
```
Comment (bziobrowski) [inline core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgGetKeywordsFunctionFactory.java:121]: Shouldn't function return more columns ? 
pg_get_keywords () → setof record ( word text, catcode "char", barelabel boolean, catdesc text, baredesc text )
see https://www.postgresql.org/docs/current/functions-info.html

File: core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgIndexFunctionFactory.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    protected PgIndexFunctionFactory(String signature) {
+        super(signature, METADATA);
+    }
+
+    @Override
+    public boolean isRuntimeConstant() {
+        return true;
+    }
+
```
Comment (bziobrowski) [inline core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgIndexFunctionFactory.java:47]: Wouldn't it be better to include all columns ?
https://www.postgresql.org/docs/current/catalog-pg-index.html

File: core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgGetKeywordsFunctionFactory.java (hunk 2 above)
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgGetKeywordsFunctionFactory.java:126]: That's not a new function, but an old one. Added missing columns in 4ff3df82553c4c551956cbdcf551cf215a5eb778

File: core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgIndexFunctionFactory.java (hunk 3 above)
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/griffin/engine/functions/catalogue/PgIndexFunctionFactory.java:47]: This one is also old. Added the missing columns in bcf6180c0d58a1e021fc3ee5f9b46b4101bf63b2

**Review by bziobrowski — APPROVED** [review]
👍

**Comment by puzpuzpuz** [comment]
@eugenels @bziobrowski thanks for your reviews!

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 294 / 305 (96.39%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/CurrentSchemasFunctionFactory.java|7|13|53.85%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgRolesFunctionFactory.java|20|21|95.24%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgLocksFunctionFactory.java|23|24|95.83%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgIndexFunctionFactory.java|28|29|96.55%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgGetKeywordsFunctionFactory.java|33|34|97.06%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgClassFunctionFactory.java|148|149|99.33%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgIsInRecoveryFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/TxIDCurrentFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgGetPartKeyDefFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgClassFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedCurrentSchemasFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgRolesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgDatabaseFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgGetKeywordsFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedTxIDCurrentFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgDatabaseFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgGetSITExprFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgGetSITExprFunctionFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgLocksFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgGetPartKeyDefFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PgIsInRecoveryFunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgIndexFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgNamespaceFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/PrefixedPgAttributeFunctionFactory.java|2|2|100.00%|

**Pull request merged**